build.gradle

```diff
@@ -320,6 +320,62 @@ subprojects {
 
   if (shouldPublish) {
 
+    if (shouldPublishWithShadow) {
```

Comment:
This is part of the "subprojects" block, which applies to all subprojects, not just clients. Since the "shadowJar" is specific to the clients, I think it should stay where it is.

The "project.shadow.component(mavenJava)" stanza (which is also clients specific) should be moved down to the clients-specific block. Schematically:

```
subprojects {
  if (clients) {
    // clients jar + publishing
  } else {
    // normal publishing
  }
}
clients {
 // nothing
}
```
should be

```
subprojects {
  if (!clients) {
    // normal publishing
  }
}
clients {
  // clients jar + publishing
}
```
An alternative could be a second `subprojects` block and moving all of the publishing down there, but that also seems kinda confusing.

Reply:
Thanks for the suggestion, I've moved the clients-specific part to its corresponding block